masakari-controller/controller/masakari_util.py

```diff
@@ -507,722 +511,202 @@ class RecoveryControllerUtilApi(object):
+            'project_id': project_id,
+            'user_domain_name': self.rc_config.conf_nova['domain'],
+            'project_domain_name': self.rc_config.conf_nova['domain'],
+            }
 
-            # Set nova_curl_method
-            nova_curl_method = "GET"
-            # Set nova_variable_url
-            nova_variable_url = "/servers/" + uuid
-            # Set nova_body
-            response_code, rbody = self._nova_curl_client(nova_curl_method,
-                                                          nova_variable_url)
+        self.auth_session = self._get_session(auth_args)
 
-        except:
+        conf_dic = self.rc_config.get_value('recover_starter')
+        api_retries = conf_dic.get('api_max_retry_cnt')
+
+        self.nova_client = nova_client.Client(self.NOVA_API_VERSION,
+                                              session=self.auth_session,
+                                              connect_retries=api_retries)
+
+    def _get_session(self, auth_args):
+        """ Return Keystone API session object."""
+        loader = loading.get_plugin_loader('password')
+        auth = loader.load_from_options(**auth_args)
+        sess = session.Session(auth=auth)
+
+        return sess
+
+    def _fetch_project_id(self):
+        auth_args = {
+            'auth_url': self.rc_config.conf_nova['auth_url'],
+            'username': self.rc_config.conf_nova['admin_user'],
+            'password': self.rc_config.conf_nova['admin_password'],
+            'project_name': self.rc_config.conf_nova['project_name'],
+            'project_domain_name': self.rc_config.conf_nova['domain'],
+            'user_domain_name': self.rc_config.conf_nova['domain'],
+            }
+        sess = self._get_session(auth_args)
 
-            self.rc_util.syslogout_ex("RecoveryControllerUtilApi_0001",
-                                      syslog.LOG_ERR)
-            error_type, error_value, traceback_ = sys.exc_info()
-            tb_list = traceback.format_tb(traceback_)
-            self.rc_util.syslogout(error_type, syslog.LOG_ERR)
-            self.rc_util.syslogout(error_value, syslog.LOG_ERR)
-            for tb in tb_list:
-                self.rc_util.syslogout(tb, syslog.LOG_ERR)
+        ks_client = keystone_client.Client(self.KEYSTONE_API_VERSION,
+                                           session=sess)
+        project_name = self.rc_config.conf_nova['project_name']
+        projects = filter(lambda x: (x.name == project_name),
+                         ks_client.projects.list())
 
-            msg = "[ nova_curl_method=" + nova_curl_method + " ]"
-            self.rc_util.syslogout(msg, syslog.LOG_ERR)
-            msg = "[ nova_variable_url=" + nova_variable_url + " ]"
-            self.rc_util.syslogout(msg, syslog.LOG_ERR)
+        msg = ("Project name: %s doesn't exist in project list."
+               % self.rc_config.conf_nova['project_name'])
+        assert len(projects) == 1, msg
```

Comment:
About the assert statement, this will raise AssertionError. This could happen if you set wrong project_name in config.
IMO, its better to raise smt such as KeyError. Plus, we might loose this logic if run with the -O flag.